circuits/cpp/src/aztec3/circuits/abis/c_bind.cpp

```diff
@@ -238,39 +195,59 @@ WASM_EXPORT void abis__compute_function_leaf(uint8_t const* function_leaf_preima
 }
 
 /**
- * @brief Compute a function tree root from its leaves.
+ * @brief Compute a function tree root from its nonzero leaves.
  * This is a WASM-export that can be called from Typescript.
  *
- * @details given a `uint8_t const*` buffer representing a function tree's leaves,
- * compute the corresponding tree's root and return the serialized results
- * in the `output` buffer.
+ * @details given a serialized vector of nonzero function leaves,
+ * compute the corresponding tree's root and return the
+ * serialized results via `root_out` buffer.
  *
- * @param function_leaves_buf a buffer of bytes representing the leaves of the function tree,
- * where each leaf is assumed to be a serialized field
- * @param num_leaves the number of leaves in leaves_buf
- * @param output buffer that will contain the output. The serialized function tree root.
+ * @param function_leaves_in input buffer representing a serialized vector of
+ * nonzero function leaves where each leaf is an `fr` starting at the left of the tree
+ * @param root_out buffer that will contain the serialized function tree root `fr`.
  */
-WASM_EXPORT void abis__compute_function_tree_root(uint8_t const* function_leaves_buf,
-                                                  uint8_t num_leaves,
-                                                  uint8_t* output)
+WASM_EXPORT void abis__compute_function_tree_root(uint8_t const* function_leaves_in, uint8_t* root_out)
 {
+    std::vector<NT::fr> leaves;
+    // fill in nonzero leaves to start
+    read(function_leaves_in, leaves);
+    // fill in zero leaves to complete tree
     NT::fr zero_leaf = FunctionLeafPreimage<NT>().hash(); // hash of empty/0 preimage
```

Comment:
Do we eventually want to have this also be 0, as we did for the NullifierTree rather than some known preimage?

Reply:
Not sure, what is the advantage of doing it either way?